include/silo/preprocessing/preprocessing_compressor.h

```diff
@@ -0,0 +1,35 @@
+#ifndef SILO_INCLUDE_SILO_PREPROCESSING_PREPROCESSING_COMPRESSOR_H_
+#define SILO_INCLUDE_SILO_PREPROCESSING_PREPROCESSING_COMPRESSOR_H_
+
+#include <oneapi/tbb/enumerable_thread_specific.h>
+#include <duckdb.hpp>
+
+namespace silo {
+class ReferenceGenomes;
+class ZstdCompressor;
+}  // namespace silo
+
+namespace silo::preprocessing {
+
+class Compressor {
+  public:
+   static void initialize(const silo::ReferenceGenomes& reference_genomes);
+
+   static void compressNuc(
```

Comment:
Do those function really need to be static? As far as I can judge this: Those function are registered as user defined functions in the particular DuckDB instance that lives in the `PreprocessingDatabase`. I.e. the function need the lifetime of the `PreprocessingDatabase`.
 -> Would it be enough to let the `PreprocessingDatabase` hold an instance of the compressor - making the compressor functions non-static?

Maybe this requires a slight refactoring such that the `preprocessing_db` in the `Preprocessor` is initialized after reading the reference genomes (because the constructor needs the reference genomes to construct and hold the compressors such that they live as long as the DuckDB instance).

Reply:
I think the problem is not with these functions, that they are static. (It is the same as having them in a namespace). But rather, that we have the static maps, which they use. 
But I like your idea. This way, we would not have to reset them. Static variables are always a problem.

Reply:
Yes sure, the static caches are the problem. But most probably, by the caches non-static, those functions here will also have to be non-static.

Reply:
In the end it's always the same story: Everything that's stored statically somewhere produces issues in testing.

Reply:
I did not want to have them static as well. The call to connection.CreateVectorizedFunction is the culprit here.

I needed to refactor it to a static function (and also those ugly static maps) because I could not register a non-static function as an UDF. Maybe you are able to?

Reply:
Ok, maybe we'll give it a shot :)